src/elf_parser/mod.rs

```diff
@@ -392,11 +395,15 @@ impl<'a> Elf64<'a> {
             .elf_bytes
             .get(string_range)
             .ok_or(ElfParserError::OutOfBounds)?;
+        let string_bytes_unparsed = match String::from_utf8(unterminated_string_bytes.to_vec()) {
```

Comment:
[As I said](https://github.com/solana-labs/rbpf/issues/532#issuecomment-1742152302), for security reasons the second search may only be done on the error path, not the general one, otherwise the entire string length limit becomes useless.

So `string_bytes_unparsed` must be moved into the `ok_or` below (which becomes an `ok_or_else`).

Reply:
@ksolana You might have forgotten to include the fix in your rebase, because it is still outside.